src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Data/SqlServer/00000000000000_CreateIdentitySchema.Designer.cs

```diff
@@ -188,47 +186,53 @@ protected override void BuildTargetModel(ModelBuilder modelBuilder)
 
             modelBuilder.Entity("Microsoft.AspNetCore.Identity.IdentityRoleClaim<string>", b =>
                 {
-                    b.HasOne("Microsoft.AspNetCore.Identity.IdentityRole")
+                    b.HasOne("Microsoft.AspNetCore.Identity.IdentityRole", null)
                         .WithMany()
                         .HasForeignKey("RoleId")
-                        .OnDelete(DeleteBehavior.Cascade);
+                        .OnDelete(DeleteBehavior.Cascade)
+                        .IsRequired();
                 });
```

Comment:
Likewise for `IsRequired` here? (The input model didn't change in these respects.)

Reply:
Not expected. Did a convention change?

Reply:
Investigation on `IsRequired`:
* `IdentityRoleClaim.RoleId` is marked as required in the model: https://github.com/aspnet/AspNetCore/blob/master/src/Identity/EntityFrameworkCore/src/IdentityDbContext.cs#L138
* This hasn't changed since 2.2: https://github.com/aspnet/AspNetCore/blob/release/2.2/src/Identity/EntityFrameworkCore/src/IdentityDbContext.cs#L138
* The type is `string`, which means required is not the default
* The column is being created as non-nullable in both 2.2 and 3.0
* So it seems to me like this was a bug in EF Core 2.2, which has now been fixed in EF Core 3.0.
* Since it doesn't impact the schema, I think we should retain `IsRequired` since as far as I can tell it is correct.